themes/V3/5ePHB/snippets/tableOfContents.gen.js

```diff
@@ -27,7 +27,31 @@ const getTOC = (pages)=>{
 		});
 	};
 
+	const add4 = (title, page)=>{
+		if(!_.last(res)) add1(null, page);
+		if(!_.last(_.last(res).children)) add2(null, page);
+		if(!_.last(_.last(_.last(res).children))) add3(null, page);
+		_.last(_.last(_.last(res).children).children).children.push({
+			title    : title,
+			page     : page + 1,
+			children : []
+		});
+	};
+
```

Comment:
this looks very readable, thinking if we can refactor this to make it more clear

Reply:
What needs clarification?

Reply:
I just think code should be readable to a point, and even knowing JS i have no idea what that code does, maybe add a simple comment about it?